### ISSUE TITLE
function rest parameters not working as expected

### ISSUE THREAD

**Issue by grey93** [6to5/6to5#405]
Say i have a ```rest.js```

```js
var sum = (first, ...numbers) => {
  let result = first,
  i = 0,
  len = numbers.length;

  while (i < len) {
    result += numbers[i];
    i++;
  }

  return result;
}

console.log(sum(1, 2, 3));
```

And I do

```shell
6to5 rest.js
```

i get,

```js
"use strict";

var sum = function (first) {
  var numbers = [];

  for (var _key = 1; _key < arguments.length; _key++) {
    numbers[$__0] = arguments[_key];
  }

  var result = first, i = 0, len = numbers.length;

  while (i < len) {
    result += numbers[i];
    i++;
  }

  return result;
};

console.log(sum(1, 2, 3));
```

which gives an error on line 7

```shell
[stdin]:7
    numbers[$__0] = arguments[_key];
            ^
ReferenceError: $__0 is not defined
```

with

```shell
6to5-node rest.js   
```

i get,

```shell
var sum = (first, ...numbers) => {
                            ^
ReferenceError: $__0 is not defined
```

but,

```shell
traceur rest.js   
```

gives the expected output ```6```.

6to5 -V ```2.6.0```

**Comment by sebmck** [comment]
Whoops, sorry about that! Fixed as of 2.6.1

**Comment by grey93** [comment]
Now it's creating

```js
"use strict";

var sum = function (first) {
  var numbers = [];

  for (var _key = 1; _key < arguments.length; _key++) {
    numbers[_key] = arguments[_key];
  }

  var result = first, i = 0, len = numbers.length;

  while (i < len) {
    result += numbers[i];
    i++;
  }

  return result;
};

console.log(sum(1, 2, 3));
```

which ends up giving ```NaN```

i guess because line 7 should be 

```js
numbers[_key - 1] = arguments[_key];
```

**Comment by sebmck** [comment]
Ugh yeah, you're right, sorry. Fixed in 2.6.1
